File: Producto_electronico.py
from abc import ABC, abstractmethod

class ProductoElectronico(ABC):
    def __init__(self, id_producto, nombre, marca, precio, stock):
        self.id_producto = id_producto
        self.nombre = nombre
        self.marca = marca
        self.precio = precio
        self.stock = stock

    @abstractmethod
    def mostrar_informacion(self):
        pass

    @abstractmethod
    def actualizar_precio(self, porcentaje):
        pass

File: ProductoElectronico.java
package SistemaGestionTienda;

public abstract class ProductoElectronico {
    
    //VARIABLES DE INSTANCIA
    private String idProducto;
    private String nombre;
    private String marca;
    private double precio;
    private int stock;
    
    // CONSTRUCTORES
    public ProductoElectronico(String idProducto, String nombre, String marca, double precio, int stock) {
        this.idProducto = idProducto;
        this.nombre = nombre;
        this.marca = marca;
        this.precio = precio;
        this.stock = stock;
    }
    // Getters y Setters (Encapsulamiento)
    public String getIdProducto() { 
        return idProducto; 
    }
    public String getNombre() { 
        return nombre; 
    }
    public String getMarca() { 
        return marca; 
    }
    public double getPrecio() { 
        return precio; 
    }
    public int getStock() { 
        return stock; 
    }

    public void setPrecio(double precio) { this.precio = precio; }
    public void setStock(int stock) { this.stock = stock; }

    // Métodos abstractos
    public abstract void mostrarInformacion();
    public abstract void actualizarPrecio(double porcentaje);
}
